http/swagger.yml

```diff
@@ -4564,6 +4566,28 @@ components:
           type: string
         dialect:
           $ref: "#/components/schemas/Dialect"
+    FileASTNode:
+      description: flux File AST node
+      type: object
+      properties:
+        type:
+          description: type of AST node
+          type: string
+          enum:
+            - File
+        package:
+          description: package declaration
+          type: object
```

Comment:
This object looks like it has these properties:

BaseNode
```json
{
    "location": {
        "file": "string",
        "start": {
            "line": 1,
            "column": 1
        },
    },
    "errors": [
        {"msg": "string"}
    ]
}
```
and

File

```json
        {
            "name": "string",
            "package": [
                {"name": "string"}
            ],
            "imports": [
                {
                    "as": {"name": "string"},
                    "path": {"value": "string"}
                }
            ],
            "body": [
                // oneOf BadStatement, VariableAssignment, MemberAssignment, ExpressionStatement, etc
            ]
        }
```

Would you define the name field and fill in the package, imports and body objects?  Best would be to create more definitions that can be referred to here.

Reply:
BaseNode is only used by the Flux parser to record source code location information. It should never be specified as part of this API which is why I didn't include it. But I can add the other fields.